File: namsor/ruby/src/CommunityEngageBatchExample.rb
require "json"
require "namsor_client"

NamsorClient.configure do |config|
    config.api_key["api_key"] = "YOUR_API_KEY"
end

personal_names_1 = NamsorClient::FirstLastNameGeoIn.new
personal_names_1.id = "id"
personal_names_1.first_name = "firstName"
personal_names_1.last_name = "lastName"
personal_names_1.country_iso2 = "countryIso2"

personal_names_2 = NamsorClient::FirstLastNameGeoIn.new
personal_names_2.id = "id"
personal_names_2.first_name = "firstName"
personal_names_2.last_name = "lastName"
personal_names_2.country_iso2 = "countryIso2"

personal_names = [
    personal_names_1,
    personal_names_2,
]

batch_first_last_name_geo_in = NamsorClient::BatchFirstLastNameGeoIn.new
batch_first_last_name_geo_in.personal_names = personal_names

begin
    response = NamsorClient::PersonalApi.new.community_engage_batch(
        {
            batch_first_last_name_geo_in: batch_first_last_name_geo_in,
        },
    )

    p response
rescue NamsorClient::ApiError => e
    puts "Exception when calling PersonalApi#community_engage_batch: #{e}"
end

File: namsor/python/src/CommunityEngageBatchExample.py
import json
from datetime import date, datetime
from pprint import pprint

from namsor_client import ApiClient, ApiException, Configuration, api, models

configuration = Configuration(
    api_key={"api_key": "YOUR_API_KEY"},
)

with ApiClient(configuration) as api_client:
    personal_names_1 = models.FirstLastNameGeoIn(
        id="id",
        firstName="firstName",
        lastName="lastName",
        countryIso2="countryIso2",
    )

    personal_names_2 = models.FirstLastNameGeoIn(
        id="id",
        firstName="firstName",
        lastName="lastName",
        countryIso2="countryIso2",
    )

    personal_names = [
        personal_names_1,
        personal_names_2,
    ]

    batch_first_last_name_geo_in = models.BatchFirstLastNameGeoIn(
        personalNames=personal_names,
    )

    try:
        response = api.PersonalApi(api_client).community_engage_batch(
            batch_first_last_name_geo_in=batch_first_last_name_geo_in,
        )

        pprint(response)
    except ApiException as e:
        print("Exception when calling PersonalApi#community_engage_batch: %s\n" % e)
